docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,406 @@
+
+### Serving community history archives
+
+Community owner nodes go through the following (high level) process to provide community members with message histories (assumes community owner node is available 24/7):
+
+1. Community owner creates a Status community (previously known as [org channels](https://github.com/status-im/specs/pull/151))
+2. Community owner doesn't disable community history archive support (on by default but can be turned off as well - see [UI feature spec](https://github.com/status-im/feature-specs/pull/36))
+3. A special type of channel to exchange metadata about the archival data is created, this channel should not be visible in the user interface
+4. Community owner invites members and creates additional channels
+5. Community owner node receives messages and stores them into local database
+6. After 7 days, the community owner node exports and compresses last 7 days worth of messages from database and bundles it together with a [message archive index](#waku-message-archive-index) into a torrent, from which it then creates a magnet link ([Magnet URI scheme](https://en.wikipedia.org/wiki/Magnet_URI_scheme), [Extensions for Peers to Send Metadata Files](https://www.bittorrent.org/beps/bep_0009.html)) 
+7. Community owner sends the magnet link created in step 6) to community members via special channel created in step 3) through the Waku network
+8. Every subsequent 7 days, the steps 6) and 7) are repeated and the new message archive data is appended to the previously created message archive data 
+
+### Serving archives for missed messages
+
+If the community owner node goes offline (where "offline" means, the owner node's main process is no longer running), it MUST go through the following process:
+
+1. Community owner node restarts
+2. Community owner node requests messages from store nodes for the missed time range for all channels in their community
+3. All missed messages are stored into community owner node's local message database
+4. If 7 or more days have elapsed since the last message history torrent was created, the community owner node will perform step 6) and 7) of [Serving community history archives](#serving-community-history-archives) for every 7 days worth of messages in the missed time range (e.g. if the node was offline for 30 days, it will create 4 message history archives)
+
+### Receiving community history archives
+
+Community member nodes go through the following (high level) process to fetch and restore community message histories:
+
+1. User joins community and becomes community member (see [org channels spec](https://github.com/status-im/specs/pull/151))
+2. By joining a community, member nodes automatically subscribe to special channel for message archive metadata exchange provided by the community
+3. Member node requests live message history (last 30 days) of special community channel from store nodes (in addition to last 7 days of message for all other community channels)
```

Comment:
```suggestion
3. Member node requests live message history (last 30 days)  of all the community channels including the special channel from store nodes
```

Reply:
Actually, this change doesn't necessarily seem to apply. AFAIK messages are generally only fetched for last 7 days for all community channels. We actually need to change this so that the range is expanded to 30 days for the special channel, to ensure community members receive the latest published magnet link.

@richard-ramos Can you confirm the 7 day rule per channel?

Reply:
This is my understanding, please let me know if the process is different than this. 
- For **Live messages**, there is no need for a magnet link, they can be fetched from store nodes (it is also mentioned at the beginning of the specs).
- Once the live messages are fetched, among them there would be some magnet links. The community member then uses the most recent magnet link and proceeds as normal (fetches the most recent 7-day archives and deletes part of its database that corresponds to the time range covered by each archive)